lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/GeneratingSuggester.java

```diff
@@ -175,67 +175,85 @@ private static int calcThreshold(String word) {
     }
 
     // suffixes
-    processFST(
-        dictionary.suffixes,
-        (key, ids) -> {
-          String suffix = new StringBuilder(toString(key)).reverse().toString();
-          if (misspelled.length() <= suffix.length() || !misspelled.endsWith(suffix)) return;
```

Comment:
We traversed the whole FST just to rule out everything not `endsWith`. FSTs can do better :)